backend/src/sms/services/sms-callback.service.ts

```diff
@@ -94,7 +114,81 @@ const parseEvent = async (req: Request): Promise<void> => {
     )
   }
 }
+
+// for future context: we only care about when the twilio callback is a finalized status
+// we don't want to update the status pre-emptively until we receive the finalized status
+// initial state --> UNSENT
+const parseTransactionalEvent = async (req: Request): Promise<void> => {
+  const {
+    MessageStatus: twilioMessageStatus,
+    ErrorCode: twilioErrorCode,
+    MessageSid: messageId,
+  } = req.body
+  if (FINALIZED_STATUS.indexOf(twilioMessageStatus as string) === -1) {
+    return
+  }
+  logger.info({
+    message: 'Updating message in sms_messages_transactional',
+    messageId,
+    action: 'parseTransactionalEvent',
+  })
+
+  if (twilioErrorCode) {
+    await SmsMessageTransactional.update(
+      {
+        errorCode: twilioErrorCode,
+        status: TransactionalSmsMessageStatus.Error,
+      } as SmsMessageTransactional,
+      {
+        where: {
+          messageId: messageId,
+        },
+      }
+    )
+  } else {
+    // longer messages are delivered in multiple segments
+    // each segment has a separate delivery status
+    // Update the message as successful only if there does not exist previous failed status
+    // update only if sent or delivered
+    // map here because twilio does not explicitly send "error"
+    const mappedStatus = mapTwilioMessageStatus(twilioMessageStatus)
+    if (
+      mappedStatus === TransactionalSmsMessageStatus.Sent ||
+      mappedStatus === TransactionalSmsMessageStatus.Delivered
+    ) {
+      await SmsMessageTransactional.update(
+        {
+          updatedAt: new Date(),
+          status: mappedStatus,
+        } as SmsMessageTransactional,
+        {
+          where: {
+            messageId: messageId,
+            errorCode: { [Op.eq]: null },
+          },
+        }
+      )
+    }
+  }
+}
+
+const mapTwilioMessageStatus = (
+  status: string
```

Comment:
similarly, would be good if this can be an `enum`, ideally imported from Twilio's library (if it exists there) and if not, it should be `enum` that we define ourselves. this will allow the `switch` case to ensure we have exhausted all the possibilities.

Reply:
this is... already an enum?

Reply:
referring to `status: string`

Reply:
same as above comment, let's do this in the twilio refactoring should the day come, it's currently stored as an array of strings so changing to enum right now wouldn't make sense unless we refactor the whole bit.